crates/core/tedge_agent/src/tedge_operation_converter/actor.rs

```diff
@@ -578,11 +607,26 @@ Action: {action}
             .unwrap();
         let operation = &self.operation;
         let cmd_id = &self.cmd_id;
+
+        let parent_operation = if cmd_id.starts_with("sub:") {
+            // Convert topic/payload: te/device/main///cmd/restart/sub:firmware_update:sub:device_profile:robot-123    {"status":"init"}
+            // into "device_profile > firmware_update > restart @ init"
+            let parents: Vec<&str> = cmd_id
+                .split("sub:")
+                .filter(|x| !x.is_empty())
+                .map(|x| x.split(':').next().unwrap())
+                .collect();
+            let parents: Vec<&str> = parents.into_iter().rev().collect();
```

Comment:
I pushed a *fixup* so only `GenericCommandState` is aware of the structure of a command id.